cirq/contrib/qasm_import/_parser.py

```diff
@@ -284,56 +286,101 @@ def p_term(self, p):
-            qubits = []
-            for num in range(self.qregs[reg]):
-                arg_name = self.make_name(num, reg)
-                if arg_name not in self.qubits.keys():
-                    self.qubits[arg_name] = NamedQubit(arg_name)
-                qubits.append(self.qubits[arg_name])
-            p[0] = qubits
-        else:
-            raise QasmException('Undefined quantum/classical register "{}" '
+        if reg not in self.qregs.keys():
+            raise QasmException('Undefined quantum register "{}" '
+                                'at line {}'.format(reg, p.lineno(1)))
+        qubits = []
+        for num in range(self.qregs[reg]):
+            arg_name = self.make_name(num, reg)
+            if arg_name not in self.qubits.keys():
+                self.qubits[arg_name] = NamedQubit(arg_name)
+            qubits.append(self.qubits[arg_name])
+        p[0] = qubits
+
+    # carg : ID
+    #     | ID '[' NATURAL_NUMBER ']'
+
+    def p_classical_arg_register(self, p):
+        """carg : ID """
+        reg = p[1]
+        if reg not in self.cregs.keys():
+            raise QasmException('Undefined classical register "{}" '
                                 'at line {}'.format(reg, p.lineno(1)))
 
+        p[0] = [self.make_name(num, reg) for num in range(self.cregs[reg])]
+
     def make_name(self, num, reg):
         return str(reg) + "_" + str(num)
 
-    def p_arg_bit(self, p):
-        """arg : ID '[' NATURAL_NUMBER ']' """
+    def p_quantum_arg_bit(self, p):
+        """qarg : ID '[' NATURAL_NUMBER ']' """
         reg = p[1]
         num = p[3]
         arg_name = self.make_name(num, reg)
-        if reg in self.qregs.keys():
-            size = self.qregs[reg]
-            if num > size - 1:
-                raise QasmException('Out of bounds qubit index {} '
-                                    'on register {} of size {} '
-                                    'at line {}'.format(num, reg, size,
-                                                        p.lineno(1)))
-            if arg_name not in self.qubits.keys():
-                self.qubits[arg_name] = NamedQubit(arg_name)
-            p[0] = [self.qubits[arg_name]]
-        else:
+        if reg not in self.qregs.keys():
             raise QasmException('Undefined quantum register "{}" '
                                 'at line {}'.format(reg, p.lineno(1)))
+        size = self.qregs[reg]
+        if num > size - 1:
+            raise QasmException('Out of bounds qubit index {} '
+                                'on register {} of size {} '
+                                'at line {}'.format(num, reg, size,
+                                                    p.lineno(1)))
+        if arg_name not in self.qubits.keys():
+            self.qubits[arg_name] = NamedQubit(arg_name)
+        p[0] = [self.qubits[arg_name]]
+
+    def p_classical_arg_bit(self, p):
+        """carg : ID '[' NATURAL_NUMBER ']' """
+        reg = p[1]
+        num = p[3]
```

Comment:
Use `index` instead of `num`, because `num` suggests lengths or counts which will make comparisons look off-by-one (e.g. an index is out of range if >= whereas a length is out of range if >).